src/background.js

```diff
@@ -216,57 +223,53 @@ chrome.storage.local.get(storedState => {
-                },
-                ['responseHeaders']
-            )
-        }
-        if(!chrome.webRequest.onHeadersReceived.hasListener(onHeadersReceivedListener)){
-            chrome.webRequest.onHeadersReceived.addListener(
-                onHeadersReceivedListener,
-                {
-                    urls: ['<all_urls>'],
-                    types: ['main_frame', 'sub_frame', 'xmlhttprequest']
-                },
-                ['blocking', 'responseHeaders']
-            )
-        }
-        if(!chrome.tabs.onActivated.hasListener(tabActivationListener)){
-            chrome.tabs.onActivated.addListener(tabActivationListener)
-        }
-        if(!chrome.tabs.onUpdated.hasListener(tabUpdateListener)){
-            chrome.tabs.onUpdated.addListener(tabUpdateListener)
-        }
+
+    if (!chrome.webRequest.onBeforeRequest.hasListener(onBeforeRequestListener)) {
+        chrome.webRequest.onBeforeRequest.addListener(
+            onBeforeRequestListener,
+            {
+                urls: ['<all_urls>'],
+                types: isFirefox() ? ['xmlhttprequest', 'imageset', 'image'] : ['image']
+            },
+            ['blocking']
+        )
+    }
+
+    if (!chrome.webRequest.onCompleted.hasListener(onCompletedListener)) {
+        chrome.webRequest.onCompleted.addListener(
+            onCompletedListener,
+            {
+                urls: ['<all_urls>'],
+                types: isFirefox() ? ['xmlhttprequest', 'imageset', 'image'] : ['image']
+            },
+            ['responseHeaders']
+        )
     }
 
-    function detachListeners(){
-        chrome.webRequest.onBeforeRequest.removeListener(onBeforeRequestListener)
-        chrome.webRequest.onCompleted.removeListener(onCompletedListener)
-        chrome.webRequest.onHeadersReceived.removeListener(onHeadersReceivedListener)
-        chrome.tabs.onActivated.removeListener(tabActivationListener)
-        chrome.tabs.onUpdated.removeListener(tabUpdateListener)
+    if (!chrome.webRequest.onHeadersReceived.hasListener(onHeadersReceivedListener)) {
+        chrome.webRequest.onHeadersReceived.addListener(
+            onHeadersReceivedListener,
+            {
+                urls: ['<all_urls>'],
+                types: ['main_frame', 'sub_frame', 'xmlhttprequest']
+            },
+            ['blocking', 'responseHeaders']
+        )
     }
 
-    if(!chrome.storage.onChanged.hasListener(updateState)){
-        chrome.storage.onChanged.addListener(updateState)
+    if (!chrome.tabs.onActivated.hasListener(onTabActivated)) {
+        chrome.tabs.onActivated.addListener(onTabActivated)
     }
-    if(!chrome.runtime.onInstalled.hasListener(checkSetup)){
-        chrome.runtime.onInstalled.addListener(checkSetup)
+
+    if (!chrome.tabs.onUpdated.hasListener(onTabUpdated)) {
+        chrome.tabs.onUpdated.addListener(onTabUpdated)
+    }
+
+    if (!chrome.storage.onChanged.hasListener(onChanged)) {
```

Comment:
Hi brian I tried yours too is good, `onChanged` here should be `onStateChanged` thanks